Dockerfile

```diff
@@ -22,8 +29,9 @@ RUN apk add --no-cache --virtual .build-deps \
         openssl-dev \
         musl-dev \
         libffi-dev \
-    && pip install -r /opt/certbot/dependency-requirements.txt \
     && pip install --no-cache-dir \
+        --requirement /opt/certbot/unhashed_requirements.txt \
```

Comment:
I don't think we should switch to the unhashed requirements here. While it probably doesn't matter much, the hashed requirements is an additional security check that the packages we obtained over HTTPS are the ones we expect. Since `certbot-auto` uses these requirements, all needed packages should be specified in `letsencrypt-auto-source/pieces/dependency-requirements.txt` and we shouldn't be installing anything that is (only) in `dev_constraints.txt`.

What if we:

1. Kept the `pip install -r /opt/certbot/dependency-requirements.txt` line.
2. Changed the command installing `acme` and `certbot` to include `--no-deps`.

The inclusion of `--no-deps` may cause things to break in the future, but it's a sanity check that all dependencies were properly installed from the hashed requirements and the second `pip` invocation wasn't secretly installing other packages.

Thoughts?